src/explore-education-statistics-common/src/modules/table-tool/components/PublicationForm.tsx

```diff
@@ -30,29 +31,68 @@ const formId = 'publicationForm';
 
 interface Props extends InjectedWizardProps {
   initialValues?: PublicationFormValues;
-  options: Theme[];
+  themes: Theme[];
   onSubmit: PublicationFormSubmitHandler;
 }
 
 const PublicationForm = ({
   initialValues = {
     publicationId: '',
   },
-  options,
+  themes,
   onSubmit,
   ...stepProps
 }: Props) => {
   const { isActive, goToNextStep } = stepProps;
 
   const [searchTerm, setSearchTerm] = useState('');
-  const lowercaseSearchTerm = searchTerm.toLowerCase();
+  const [selectedThemeId, setSelectedThemeId] = useState<string>('');
+
+  const publications = useMemo(() => {
+    if (selectedThemeId) {
+      setSearchTerm('');
+      return (
+        themes
+          .find(theme => theme.id === selectedThemeId)
+          ?.topics.flatMap(topic => topic.publications) ?? []
+      );
+    }
+    if (searchTerm) {
+      setSelectedThemeId('');
```

Comment:
Do we need to clear this here? State changes like this would be better within the event handler where the change happens and we already are doing this.